### ISSUE TITLE
Some Shortcut keys not working

### ISSUE THREAD

**Issue by thepiyush-303** [jitsi/jitsi-meet#15066]
### What happened?

To address the issue of shortcut keys (like Alt + T, Alt + B, Alt + S) not working as expected in Firefox due to the default behavior of the Alt key triggering the Menu Bar,

In Firefox, pressing the Alt key activates the Menu Bar at the top of the browser window, which then redirects subsequent keystrokes (T, B, S, etc.) to the Menu Bar menus instead of triggering the intended shortcut actions on our web application, such as controlling playback in an online video.

This behavior is built into Firefox and can interfere with custom keyboard shortcuts used in web applications. As a result, the shortcuts don't work as intended because the browser prioritizes its native menu navigation over the web application's event handling.
[Screencast from 2024-08-29 01-49-15.webm](https://github.com/user-attachments/assets/b4bd034d-95b8-44e0-b12d-dce03e14244d)

### Platform

- [ ] Chrome (or Chromium based)
- [X] Firefox
- [ ] Safari
- [ ] Other desktop browser
- [ ] Android browser
- [ ] iOS browser
- [ ] Electron app
- [ ] Android mobile app
- [ ] iOS mobile app
- [ ] Custom app using a mobile SDK

### Browser / app / sdk version

129.0.2 (64-bit)

### Reproducibility

- [X] The problem is reproducible on meet.jit.si

**Comment by damencho** [comment]
> To address the issue of shortcut keys (like Alt + T, Alt + B, Alt + S) not working as expected in Firefox due to the default behavior of the Alt key triggering the Menu Bar,

The shortcuts in jitsi-meet do not require Alt, press `t` or `s`. I just tested those in Firefox and everything works as expected.

**Comment by damencho** [comment]
Please, when you have questions or problems use the community forum before opening new issues, thank you.

**Comment by thepiyush-303** [comment]
Hello @damencho ,
In the future, I will make sure to consult the community forum before creating any issue. I forget to mention that this particular issue is observed only on Ubuntu, while it is working perfectly on macOS. The problem seems to be specific to the Ubuntu environment, and I would appreciate any guidance or suggestions on resolving this discrepancy.

**Comment by damencho** [comment]
So you say that invoking the shortcuts without Alt is not working? Just pressing the single letters?

**Comment by thepiyush-303** [comment]
No, the issue is specifically with the Alt key in combination with the shortcuts. When using shortcuts like Alt + T, Alt + B, and Alt + S in Firefox on Ubuntu, the Alt key triggers the Menu Bar, causing the shortcut actions not to execute on the application but execute on the menubar like in the screencast I provided above , However other shortcuts like Alt + C , Alt + L , Alt + O and all other single letters shortcuts are working perfectly fine.

![Screenshot from 2024-08-29 18-17-39](https://github.com/user-attachments/assets/39c4e07c-ea5f-4b0a-b150-09155924be03)

**Comment by thepiyush-303** [comment]
Hey @damencho ,
What if we change the ShortcutChar letter .

**Comment by thepiyush-303** [comment]
Like Alt + T, Alt + B, and Alt + S only these are causing an issue ,can we change  'T' , 'B' and 'S' with different letter?

**Comment by damencho** [comment]
That is not an option. I guess the option will be to implement a configuration to be able to change those, so anyone who finds it problematic to be able to adjust it.

**Comment by manzil-infinity180** [comment]
@damencho overriding the shortcut can be a possible or convenient way to do it I guess?

**Comment by thepiyush-303** [comment]
Hello @damencho 
can I give it a try to fix this issue?

**Comment by damencho** [comment]
Ok, just make sure you don't change the current behavior.

**Comment by yosifshaban6** [comment]
> ### What happened?
> To address the issue of shortcut keys (like Alt + T, Alt + B, Alt + S) not working as expected in Firefox due to the default behavior of the Alt key triggering the Menu Bar,
> 

@damencho @thepiyush-303 

Hello Everyone,

I tested this issue with the following environment:
OS: Ubuntu 24.04
Browser: Firefox 148.0.2

I tested the shortcuts in Firefox and noticed that the issue occurs because Firefox captures the **Alt** key to activate the browser menu bar.

However, when using Ctrl + Alt instead (for example Ctrl + Alt + T), the shortcut works correctly and the event is handled by the application.

This suggests the behavior is caused by Firefox's default menu handling rather than the application itself. Perhaps using `Ctrl + Alt` combinations for Firefox could be a possible workaround.


![Image](https://github.com/user-attachments/assets/93b5aac1-1ec7-4835-bda5-61ec759f1c44)

**Comment by Yash1927** [comment]
@damencho 

I reproduced this on the current Jitsi Meet master using Firefox on Ubuntu under WSLg. Alt+T, Alt+B, and Alt+S are intercepted by Firefox, while Alt+C, Alt+L, Alt+O, and Alt+H work. For example, Alt+T activates Firefox’s Tools menu and moves focus away from the meeting. Other reaction shortcuts then stop working until the meeting page is clicked again. Alt+T is intermittent depending on the current browser-menu and page-focus state.

I also reviewed the closed stale PR #17060. Would a clean PR adding configurable reaction shortcut mappings while preserving the existing defaults be welcome? I can include focused tests and avoid unrelated changes.

**Comment by damencho** [comment]
Sounds good

**Comment by Yash1927** [comment]
@damencho 

Thanks! I checked the current keyboard shortcut implementation and the existing Shortcuts tab in Settings. The tab currently lets users enable or disable all shortcuts and displays a read-only list, but it does not provide reaction-specific options.

I can see three possible approaches:

1. Add a deployment-level configuration that administrators or embedders can use to replace reaction shortcuts.

2. Register Ctrl+Alt reaction shortcuts as additional aliases for everyone.

3. Add a user-level setting that lets each user enable the Ctrl+Alt aliases.

I think the third approach would give users control while preserving the current behavior.
The proposed behavior would be:
- Add an “Enable Ctrl+Alt reaction shortcut alternatives” checkbox to the existing Keyboard Shortcuts settings tab.
- Keep this option disabled by default.
- When it is disabled, the existing shortcuts continue working exactly as they do now:
  - Alt+T sends 👍
  - Alt+B sends 🙁
  - Alt+S sends 😶
- When the user enables it, the Ctrl+Alt combinations are registered as additional aliases:
  - Alt+T and Ctrl+Alt+T both send 👍
  - Alt+B and Ctrl+Alt+B both send 🙁
  - Alt+S and Ctrl+Alt+S both send 😶
- The preference would be stored locally in that user’s browser, so it would not require a server administrator or meeting owner.
- The Shortcuts tab would display both combinations while the option is enabled.
- The user could turn the option off at any time to return to the original shortcuts only.

This would not change or remove any existing shortcut. It would only allow an individual Firefox user to enable additional combinations that are not intercepted by Firefox.

**Comment by damencho** [comment]
Yeah 3d approach, and maybe show that feature only for users using Firefox.
